scapy/contrib/ikev2.py

```diff
@@ -748,7 +748,7 @@ class IKEv2_Notify(IKEv2_Payload):
         ShortEnumField("type", 0, IKEv2NotifyMessageTypes),
         XStrLenField("SPI", "", length_from=lambda pkt: pkt.SPIsize),
         ConditionalField(
-            XStrLenField("notify", "", length_from=lambda pkt: pkt.length - 8),
+            XStrLenField("notify", "", length_from=lambda pkt: pkt.length - 8 - pkt.SPIsize),
```

Comment:
See the analogous calculation for `ISAKMP_payload_Notify` (line 489):

https://github.com/secdev/scapy/blob/047be325fe7b67fc05287632e7b14bc9e5d6701f/scapy/layers/isakmp.py#L480-L490